interview_cake/buy_stocks/jessicas_solution/making_money.py

```diff
@@ -0,0 +1,44 @@
+def max_profit(stock_prices):
+    """
+    what is the max profit possible between two prices
+    of stock assuming you must buy before you can sell
+    in O(n)^2 time
+    """
+    max_profit = 0
+
+    for index, current_price in enumerate(stock_prices):
+        current_index = index
+        max_index = len(stock_prices)
+
+        for i in range(current_index+1, max_index):
+            selling_profit = stock_prices[i] - current_price
+
+            if selling_profit > max_profit:
+                max_profit = selling_profit
+
+    return max_profit
+
+def max_profit_faster(stock_prices):
+    """
+    make money faster in O(n) time
+    buy low then sell high later for max profits
+    """
+    if len(stock_prices) is 0:
+        return 0
+
+    open_price = stock_prices[0]
+    low_price = open_price
+    high_price = open_price
+    max_profit = high_price - low_price
```

Comment:
yeah I was thinking about this. It was not necessary to set these vars in this way.  I was debating it because when the stock market opens the only value the low and high can be is the opening price.  So I thought it would make more sense in the real world, but sounds like it is actually just confusing. So I will change it. thanks!